synapse/http/client.py

```diff
@@ -1057,11 +1057,17 @@ def dataReceived(self, incoming_data: bytes) -> None:
         if not self.parser:
 
             def on_header_field(data: bytes, start: int, end: int) -> None:
-                if data[start:end] == b"Location":
+                if data[start:end] == b"Location" or data[start:end] == b"location":
```

Comment:
```suggestion
                if data[start:end].lower() == b"location":
```
Whilst maybe unconventional, it's valid to pass headers in all uppercase. Generally it's better to perform header matching in a case-insensitive manner

Reply:
Thanks for pointing that out - I was worried about calling `lower()` on non-string bytes and causing an error but thinking about it for a moment the code is in the `on_header_field` function so that shouldn't happen if the parser is doing its job.